src/Credentials/ImpersonatedServiceAccountCredentials.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Google\Auth\Credentials;
+
+use Google\Auth\Iam;
+use Google\Auth\IAMSignerTrait;
+use Google\Auth\SignBlobInterface;
+
+class ImpersonatedServiceAccountCredentials extends UserRefreshCredentials implements SignBlobInterface
```

Comment:
Rather than extend `UserRefreshCredentials`, I think it would be better to create these in the Constructor. e.g.

```php
$this->sourceClient = new UserRefreshCredentials($scope, $jsonKey['source_credentials']);
```

Then we can make this just extend the `CredentialsLoader` class like the other credential types:
```suggestion
class ImpersonatedServiceAccountCredentials extends CredentialsLoader implements SignBlobInterface
```

We would then need to define `fetchAuthToken`, `getCacheKey`, and `getLastReceivedToken` as follows:

```php

    /**
     * @param callable $httpHandler
     *
     * @return array<mixed> {
     *     A set of auth related metadata, containing the following
     *
     *     @type string $access_token
     *     @type int $expires_in
     *     @type string $scope
     *     @type string $token_type
     *     @type string $id_token
     * }
     */
    public function fetchAuthToken(callable $httpHandler = null)
    {
        return $this->sourceCredentials->fetchAuthToken($httpHandler);
    }

    /**
     * @return string
     */
    public function getCacheKey()
    {
        return $this->sourceCredentials->getCacheKey();
    }

    /**
     * @return array<mixed>
     */
    public function getLastReceivedToken()
    {
        return $this->sourceCredentials->getLastReceivedToken();
    }

```